src/ImageSharp/Metadata/Profiles/Exif/ExifReader.cs

```diff
@@ -119,7 +110,30 @@ internal abstract class BaseExifReader
 
         public bool IsBigEndian { get; protected set; }
 
-        protected abstract void RegisterExtLoader(uint offset, Action loader);
+        public List<(ulong offset, ExifDataType dataType, ulong numberOfComponents, ExifValue exif)> BigValues { get; } = new ();
+
+        protected void ReadBigValues(List<IExifValue> values)
+        {
+            ulong maxSize = 0;
+            foreach ((ulong offset, ExifDataType dataType, ulong numberOfComponents, ExifValue exif) in this.BigValues)
+            {
+                ulong size = numberOfComponents * ExifDataTypes.GetSize(dataType);
+                if (size > maxSize)
+                {
+                    maxSize = size;
+                }
+            }
+
+            Span<byte> buf = new byte[maxSize];
```

Comment:
```suggestion
            byte[] bufferToReturnToPool = null;
            Span<byte> buffer = (maxSize <= 256    // better to use a defined constant for this
                ? stackalloc byte[256]
                : bufferToReturnToPool = ArrayPool<byte>.Shared.Rent(maxSize)
            ).Slice(maxSize);

            // ...

            if (bufferToReturnToPool is not null)
            {
                ArrayPool<byte>.Shared.Return(bufferToReturnToPool);
            }
```
To avoid the allocation, and try to use stack allocated buffer.

Or can we assume that `maxSize <= 256` will be true for all inputs? If so, the code can be simplified to just stack allocated and then slice the buffer*.

\* `stackalloc byte[ConstantWithPowerOf2]` is cheaper than `stackalloc byte[variable]`.

Reply:
@JimBobSquarePants what should we do in such cases?
`MemoryAllocator` dragging to here is probably impractical.

Reply:
Do we have any numbers for how often `maxSize` is over 256?

I would always fallback to the MemoryAllocator rather than the `ArrayPool`.

Reply:
A few, mostly it's relevant for `StripOffsets`, `StripByteCounts`. Examples, for ~4Mb file and 8Kb (recommended) strip size, required about 512 strips * 4 (long size) = 2Kb bytes for stripOffsets tag value, 16Mb - 8Kb, extreme 1Gb (BigTiff can greater than 4Gb) - 512Kb

Reply:
If I'm reading this correctly we should just use the MemoryAllocator. We'll be switching out for the new one soon enough anyway.